core/ibft_test.go

```diff
@@ -1911,6 +1911,201 @@ func TestIBFT_ValidPC(t *testing.T) {
 		assert.False(t, i.validPC(certificate, rLimit, 0))
 	})
 
+	t.Run("different rounds in prepare messages", func(t *testing.T) {
```

Comment:
This test doesn't cover the line it should be covering:
```go
	// Make sure all the messages have the same round
	if !messages.AllHaveSameRound(allMessages) {
		return false
	}
```

It's because you've set the round for all messages to be _below_ `rLimit`, instead of equal or above, and thus they are being caught in another condition check (which is covered by a different unit test already).

Reply:
Fixed

[fa90319](https://github.com/madz-lab/go-ibft/pull/10/commits/fa903198182ec1e95d242d3f0591f771b2851d97)